### PR TITLE
Add ability to unselect frame in the performance page

### PR DESCRIPTION
**Pull request by kenzieschmoll**
unselect-frame
Fixes https://github.com/flutter/devtools/issues/2766

### CODE REVIEW

File: packages/devtools_app/lib/src/performance/performance_controller.dart [hunk 1]
```diff
@@ -192,9 +192,17 @@ class PerformanceController
   final _displayRefreshRate = ValueNotifier<double>(defaultRefreshRate);
 
   Future<void> selectFrame(FlutterFrame frame) async {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/performance/performance_controller.dart:194]: method name is now misleading. Maybe it is now "clickFrame" or "toggleFrame"

File: packages/devtools_app/lib/src/performance/timeline_flame_chart.dart [hunk 2]
```diff
@@ -1036,7 +1036,12 @@ class SelectedFrameBracketPainter extends FlameChartPainter {
 
   @override
   void paint(Canvas canvas, Size size) {
-    if (selectedFrame == null) return;
+    if (selectedFrame == null) {
+      canvas
```
Comment (jacob314) [inline packages/devtools_app/lib/src/performance/timeline_flame_chart.dart:1040]: why this save and restore? that should be a no-op.

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/performance/timeline_flame_chart.dart:1040]: you cant call `restore` on Canvas without first calling `save`. It throws an exception if you do

File: packages/devtools_app/lib/src/performance/performance_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/performance/performance_controller.dart]: changed

File: packages/devtools_app/lib/src/performance/timeline_flame_chart.dart (hunk 2 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/performance/timeline_flame_chart.dart:1046]: right but why are you doing a save + restore with nothing in the middle?
That seem like writing:
```dart
foo += 1;
foo -= 1;
```

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/performance/timeline_flame_chart.dart:1046]: this is to ensure that the selected frame painter (the blue brackets that surround the frame's timeline events) disappears when a frame is unselected. We essentially need to repaint an empty scene for this custom painter.

**Review by jacob314 — APPROVED** [review]
lgtm
with the save+restore removed.

File: packages/devtools_app/lib/src/performance/timeline_flame_chart.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/performance/timeline_flame_chart.dart]: hmm - maybe this was a reloading issue (or operator error) when I saw this behavior earlier. Was able to remove the canvas calls and still get the desired effect.

**Pull request merged**